service/worker/fx.go

```diff
@@ -54,6 +56,10 @@ var Module = fx.Options(
 	deletenamespace.Module,
 	scheduler.Module,
 	batcher.Module,
+	dlq.Module,
+	fx.Provide(func(c resource.HistoryClient) dlq.HistoryServiceClient {
```

Comment:
it feels weird to do this conversion here.. only the dlq package should have to care about its limited interface. why not do it in dlq.newComponent? and yeah, call the constructor in the test instead of the module, but is that really so bad?

Reply:
I agree that this feels weird to do the conversion here. Let me provide some rationale as to why I opted for this approach:

1. **Dependency Management:** The worker package already depends on both the dlq and resource packages. However, by handling the conversion here, we prevent the dlq package from needing a dependency on the resource package.
1. **Module Awareness:** Introducing the conversion inside the dlq package would make it too cognizant of its surroundings, specifically the fact that a neighboring module provides a resource.HistoryClient. This makes the dlq module too aware and also ties it to an assumption that may change over time: a change to whether we provide the resource.HistoryClient or some other type that also implements dlq.HistoryServiceClient here would necessitate a change in the dlq module, which I would see as an unexpectedly far-reaching change.
1. **Interface Versatility:** By doing the conversion here, we preserve the versatility of the dlq module. If the conversion were in the dlq package, anyone wishing to use the module would need to provide a resource.HistoryClient. This potentially adds overhead, especially in tests where a mock resource.HistoryClient would have numerous unimplemented methods. Moreover, if we ever expand the dlq.HistoryServiceClient interface, we'd be hit with runtime errors instead of compiler errors for unimplemented methods.

In essence, the decision was a trade-off: We opted for keeping the dlq module as unaware as possible, even if it meant making the worker module slightly more complex.